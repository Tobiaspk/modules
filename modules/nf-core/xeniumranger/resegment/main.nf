process XENIUMRANGER_RESEGMENT {
    tag "$meta.id"
    label 'process_high'

    container "nf-core/xeniumranger:3.0.1"

    input:
    path(xenium_bundle)

    output:
    path("outs/**"), emit: outs
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "XENIUMRANGER_RESEGMENT module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    def args = task.ext.args ?: ""
    def prefix = task.ext.prefix ?: "${meta.id}"

    def expansion_distance = expansion_distance ? "--expansion-distance=\"${expansion_distance}\"": ""
    def resegment_nuclei = resegment_nuclei ? "--resegment-nuclei=\"${resegment_nuclei}\"": ""
    def dapi_filter = dapi_filter ? "--dapi-filter=\"${dapi_filter}\"": ""

    // Do not use boundary stain in analysis, but keep default interior stain and DAPI
    def boundary_stain = boundary_stain ? "--boundary-stain=disable": ""
    // Do not use interior stain in analysis, but keep default boundary stain and DAPI
    def interior_stain = interior_stain ? "--interior-stain=disable": ""

    """
    xeniumranger resegment \\
        --id="${prefix}" \\
        --xenium-bundle="${xenium_bundle}" \\
        --localcores=${task.cpus} \\
        --localmem=${task.memory.toGiga()} \\
        ${expansion_distance} \\
        ${resegment_nuclei} \\
        ${dapi_filter} \\
        ${boundary_stain} \\
        ${interior_stain} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        xeniumranger: \$(xeniumranger -V | sed -e "s/xeniumranger-/- /g")
    END_VERSIONS
    """

    stub:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "XENIUMRANGER_RESEGMENT module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    """
    mkdir -p outs/
    touch outs/fake_file.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        xeniumranger: \$(xeniumranger -V | sed -e "s/xeniumranger-/- /g")
    END_VERSIONS
    """
}
