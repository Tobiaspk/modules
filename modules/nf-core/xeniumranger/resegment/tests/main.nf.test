nextflow_process {

    name "Test Process XENIUMRANGER_RESEGMENT"
    script "../main.nf"
    process "XENIUMRANGER_RESEGMENT"

    tag "modules"
    tag "modules_nfcore"
    tag "xeniumranger"
    tag "xeniumranger/resegment"

    test("xeniumranger resegment") {

        when {

            process {
                """
                input[0] = path(params.test_data['mus_musculus']['10xgenomics']['xeniumranger']['xenium_bundle'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.outs.get(0).get(1).findAll { file(it).name !in [
                        'analysis_summary.html',
                        'cells.csv.gz',
                        'cells.parquet',
                        'cells.zarr.zip',
                        'transcripts.csv.gz',
                        'transcripts.parquet',
                        'transcripts.zarr.zip',
                        'analysis.zarr.zip',
                        ]}
                    ).match()
                },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'analysis_summary.html' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'cells.csv.gz' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'cells.parquet' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'cells.zarr.zip' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'transcripts.csv.gz' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'transcripts.parquet' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'transcripts.zarr.zip' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'analysis.zarr.zip' }).exists() },
            )
        }
    }

    test("xeniumranger resegment -stub") {

        options "-stub"

        when {

            process {
                """
                input[0] = path(params.test_data['mus_musculus']['10xgenomics']['xeniumranger']['xenium_bundle'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}