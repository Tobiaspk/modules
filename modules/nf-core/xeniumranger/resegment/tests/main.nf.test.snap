{
    "xeniumranger resegment stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_run"
                        },
                        "fake_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4671141281357e0ce26d9cb35fed23a8"
                ],
                "outs": [
                    [
                        {
                            "id": "test_run"
                        },
                        "fake_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4671141281357e0ce26d9cb35fed23a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-22T15:58:15.266114"
    },
    "xeniumranger resegment": {
        "content": [
            [
                "versions.yml:md5,4671141281357e0ce26d9cb35fed23a8"
            ],
            [
                "clusters.csv:md5,dc81c4f477e1a80b693818f7bfd3d8bb",
                "clusters.csv:md5,7acdd6883a131ab93188ddbae46a876b",
                "clusters.csv:md5,c50cc9ff364548abbf350d4855184fd2",
                "clusters.csv:md5,e3baf175f56e18c414b77ffea0e3f68e",
                "clusters.csv:md5,431fbab57e5bce8e75487b1a23a20a04",
                "clusters.csv:md5,26c51a9ffb3747c15feef9709b5053cc",
                "clusters.csv:md5,217d240dd587eec8814770bb31db8b95",
                "clusters.csv:md5,701edc649ec0f894154bfc1cc8c4c3e4",
                "clusters.csv:md5,d6c183c0491cdba01a91a77fb826a2f7",
                "clusters.csv:md5,12656931988f612717909b9c41ba56f5",
                "differential_expression.csv:md5,7c66ba72a51ca562e8a47f3ef0acfd2d",
                "differential_expression.csv:md5,e10a431911c1d9a7d5a231d0837153a3",
                "differential_expression.csv:md5,897666356892864b17c8d3881b3471cb",
                "differential_expression.csv:md5,083df9f6b72d6101a8f318034ed725f2",
                "differential_expression.csv:md5,d2f8ffc1bd1141361f3e839997bd58b4",
                "differential_expression.csv:md5,d0d62620461c445eb2abccdbf9aedf26",
                "differential_expression.csv:md5,3c0f1228921219cd8b16c175ac287f7f",
                "differential_expression.csv:md5,1368d8e4c302c435ba794ccb7e398952",
                "differential_expression.csv:md5,5d128e956c36d0e79cf34a8944e11e0b",
                "differential_expression.csv:md5,084a2e6a4360d5e663df624f6ef732ab",
                "components.csv:md5,0c53cf8c027928fd2a4999ef7bfc8acc",
                "dispersion.csv:md5,e8b1abb880ece8fb730ce34a15f958b4",
                "features_selected.csv:md5,c5e32d69f001f938ed316d2108a21e00",
                "projection.csv:md5,930a9e650f3164540ae251a939e4adc8",
                "variance.csv:md5,7d803a2c24f49b699f73728226ee0451",
                "projection.csv:md5,a69578a8549513482d66940b62532670",
                "cell_boundaries.csv.gz:md5,8b4f2aa455a6fb14b2669a42db32ea7e",
                "cell_boundaries.parquet:md5,e55d6a7fbec336103994baad8c8e4a9a",
                "cell_feature_matrix.h5:md5,96cb400f1b1dd6f8796daea0ad5c74e6",
                "cell_feature_matrix.zarr.zip:md5,68cfcd0624bd9305f08584f7dde5e9d2",
                "barcodes.tsv.gz:md5,04ea06796d6b28517c288904ca043582",
                "features.tsv.gz:md5,7862242129681900a9cc4086dc83b62e",
                "matrix.mtx.gz:md5,489f86fbd8d65d6b973bb9cc7c5a76f1",
                "experiment.xenium:md5,0fda66bbd54f616aa4b8f1d5290b311f",
                "gene_panel.json:md5,8890dd5fd90706e751554ac3fdfdedde",
                "metrics_summary.csv:md5,9f1269bc5f34b000945fc2070c4f1fb1",
                "morphology.ome.tif:md5,6b65fff28a38a001b8f25061737fbf9b",
                "morphology_focus_0000.ome.tif:md5,90e796ad634d14e62cf2ebcadf2eaf98",
                "nucleus_boundaries.csv.gz:md5,e417b6e293298870956d42c7106cbd0c",
                "nucleus_boundaries.parquet:md5,bacbfc3c2e956d899e1d8ccba5dd7c5e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-22T15:58:01.632344"
    }
}